examples/angular/src/app/pages/pages.component.ts

```diff
@@ -1,63 +1,128 @@
 import {
   Component,
   DestroyRef,
-  InjectionToken,
-  OnDestroy,
   OnInit,
   inject,
   signal,
 } from '@angular/core';
 import { takeUntilDestroyed } from '@angular/core/rxjs-interop';
-import { ActivatedRoute } from '@angular/router';
+import { ActivatedRoute, Router } from '@angular/router';
+import { NavigationEnd } from '@angular/router';
+import { delay, filter, startWith, tap } from 'rxjs/operators';
 
+import { DYNAMIC_COMPONENTS } from '../utils';
+
+import {
+  DotcmsLayoutComponent,
+  DotcmsNavigationItem,
+  DotCMSPageAsset,
+} from '@dotcms/angular';
+import { switchMap } from 'rxjs/operators';
+
+import { ErrorComponent } from './components/error/error.component';
+import { LoadingComponent } from './components/loading/loading.component';
 import { HeaderComponent } from './components/header/header.component';
-import { FooterComponent } from './components/footer/footer.component';
 import { NavigationComponent } from './components/navigation/navigation.component';
+import { FooterComponent } from './components/footer/footer.component';
+import { PageService } from './services/page.service';
 
-import { DYNAMIC_COMPONENTS } from '../utils';
+export type PageError = {
+  message: string;
+  status: number;
+};
 
-import { DotcmsLayoutComponent } from '@dotcms/angular';
-import { JsonPipe } from '@angular/common';
-import { DOTCMS_CLIENT_TOKEN } from '../client-token/dotcms-client';
+type PageRender = {
+  page: DotCMSPageAsset | null;
+  nav: DotcmsNavigationItem | null;
+  error: PageError | null;
+  status: 'idle' | 'success' | 'error' | 'loading';
+};
+
+
+export type PageApiOptions = {
+  depth: number;
+};
```

Comment:
if you are exporting maybe centralize it to make it less spaghetti while explaining how to use it

Reply:
Actually this needs to go away. Good catch.